src/raftstore/store/worker/region.rs

```diff
@@ -248,31 +258,177 @@ impl SnapContext {
         }
 
         timer.observe_duration();
+        region
     }
 
-    fn handle_destroy(&mut self, region_id: u64, start_key: Vec<u8>, end_key: Vec<u8>) {
-        info!(
-            "[region {}] deleting data in [{}, {})",
-            region_id,
-            escape(&start_key),
-            escape(&end_key)
-        );
+    fn handle_destroy(
+        &mut self,
+        region_id: u64,
+        start_key: Vec<u8>,
+        end_key: Vec<u8>,
+        use_delete_files: bool,
+    ) {
+        if use_delete_files {
+            if let Err(e) = util::delete_all_files_in_range(&self.kv_db, &start_key, &end_key) {
```

Comment:
if no error, why do we still call delete_all_in_range later?

Reply:
Because if an sst has only half the data in this range, then the sst will not be deleted, also half the data will not be deleted. @huachaohuang

Reply:
Yep, that's true.